### ISSUE TITLE
oandbackup scheduler won't run

### ISSUE THREAD

**Issue by protist** [jensstein/oandbackup#136]
I've set the scheduler to run, but it just doesn't work. However, if I launch oandbackup, and leave it in the foreground, then it *does* run. I guess Android is killing the oandbackup background process for some reason. I'm happy to troubleshoot more, but not sure where to start.

**Comment by jensstein** [comment]
have you tried the apk linked in issue 133 (https://github.com/jensstein/oandbackup/issues/133)?
https://drive.google.com/file/d/0B0bNUNWd2wPPdGJlVGtTNjlwc0k/view?usp=sharing
scheduling has just had some significant changes.

if you have tried that version, please send me a logcat with a schedule you were expecting to run.

**Comment by protist** [comment]
No, I haven't. I'll give that a try and report back. Cheers.

**Comment by protist** [comment]
I've not investigated logcat yet, but I've done some other troubleshooting. I installed the apk that you linked, but this hasn't solved the problem.

In general, scheduled backups still don't run. However, if I leave oandbackup running as the foreground app, scheduled backups work as expected.

I use [OS Monitor](https://play.google.com/store/apps/details?id=com.eolwral.osmonitor) to look at running processes. If I launch oandbackup, then switch only a few times to other apps, I can still detect oandbackup's process running. In this case, the scheduled backups work as expected.

However, most of the time the backgrounded oandbackup process is killed fairly rapidly. In this case (in my limited testing), no scheduled backup runs. Must the oandbackup process be running for scheduled backups to work?

**Comment by jensstein** [comment]
no, oab doesn't have to be running when the schedule is supposed to start. it uses android's alarm manager system, so the process is supposed to be started only when the time comes.
so when you have time could you send me a logcat where you were expecting a scheduled backup and tell me which android version you are using? the logcat will hopefully indicate the reason why the oab process is killed.

**Comment by protist** [comment]
Hm…  I haven't looked at logcat again yet (please let me know if it's still useful to you), but I think I've identified the problem.

I'm running Cyanogenmod 13.0 (Android 6.0.1), and I think it's intentionally supressing the scheduler. I can go to `Settings` > `Battery` > `Menu` > `Battery optimisation` > `All apps`, then select oandbackup and `Don't optimise`. The scheduler now runs fine.

FWIW I thought of testing this because [DAVdroid](https://davdroid.bitfire.at/) popped up a dialog suggesting this option might prevent their app from syncing correctly, although I'm not sure if the [arguments](https://davdroid.bitfire.at/faq/entry/sync-intervals-doze-mode/) apply to oandbackup. (More info on the request in their [code](https://gitlab.com/search?utf8=%E2%9C%93&search=battery&group_id=&project_id=447145&search_code=true&repository_ref=master).) Regardless, oandbackup has worked for three days in a row now, so I *think* it's probably fixed.

**Comment by jensstein** [comment]
that's a brilliant find :)
i'll look into whether we can use whatever davdroid does here also.

**Comment by lupa18** [comment]
Here appears a negative number and then scheduled backup don't work.
Also using CM13 and OAB 0.2.11
